### PR TITLE
Adding today button.

### PR DESCRIPTION
**Pull request by manolo**
Connected to #40

[Review on Reviewable](https://reviewable.io/reviews/vaadin/vaadin-date-picker/44)

### CODE REVIEW

**Comment by samiheikki** [comment]
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[test/overlay.html, line 57 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KAsRii3HdcGtfGx404l:-KAsRii4XWkLT5vyvVKU:-323735099) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/3fb012c10cb809f2a34ffe30115888a91709f6ec/test/overlay.html#L57)):*
This test seems to fail on my local test and in Travis also.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44)*

**Comment by tomivirkki** [comment]
The feature is missing asciidoc

---

Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[vaadin-date-picker-overlay.html, line 44 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KAsSed7R9Pt3XjrWSFE:-KAsSed7R9Pt3XjrWSFF:-268721729) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/3fb012c10cb809f2a34ffe30115888a91709f6ec/vaadin-date-picker-overlay.html#L44)):*
Since iron-icon/iron-icons is explicitly used, it should be explicitly imported as well instead of relying on transitive dependencies.

---

*[vaadin-date-picker-overlay.html, line 94 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KAsUDbccbECQtGR7VNu:-KAsUDbccbECQtGR7VNv:-1698316873) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/3fb012c10cb809f2a34ffe30115888a91709f6ec/vaadin-date-picker-overlay.html#L94)):*
This feature was planned to animate (ease-in in 1/3 seconds or so) into the current date. scrollToDate could be modified to accept a second parameter (animate) for this. Can also be done in a separate PR

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-:-KAsSv3j4UR2FCjxih6m:-1101692711)*

File: vaadin-date-picker-overlay.html [hunk 1]
```diff
@@ -18,17 +19,32 @@
...
         background: #eee;
         @apply(--shadow-elevation-2dp);
+        @apply(--layout-horizontal);
       }
 
       #scroller {
         height: 100%;
       }
+
+      #yearLabl {
```
Comment (jounik) [inline vaadin-date-picker-overlay.html]: "yearLabl" seems like a typo. Make it either "yearLabel" or "yearLbl". I prefer the first option. I would also make "todayBtn" into "todayButton".

File: vaadin-date-picker-overlay.html [hunk 2]
```diff
@@ -18,17 +19,32 @@
...
+        color: var(--primary-color, #00B4F0);
+      }
     </style>
 
-    <div id="toolbar">[[_yearAfterXMonths(_visibleMonthIndex)]]</div>
+    <div id="toolbar">
+      <div id="yearLabl">[[_yearAfterXMonths(_visibleMonthIndex)]]</div>
+      <paper-button id="todayBtn" on-tap="_scrollToCurrentMonth">
+        <iron-icon icon="event"></iron-icon>
+        TODAY
```
Comment (jounik) [inline vaadin-date-picker-overlay.html]: Don’t capitalize strings in the source, let CSS do that instead.

**Comment by manolo** [comment]
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[test/overlay.html, line 57 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KAsRii3HdcGtfGx404l:-KB6jLBjq4bEF6g-6kDp:-1503179800) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/3fb012c10cb809f2a34ffe30115888a91709f6ec/test/overlay.html#L57)):*
Renaming vars broke the test. Fixed.

---

*[vaadin-date-picker-overlay.html, line 32 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KAp7f58IzppFXu6Ra3o-r1-32:-KB6_m2PQvLslhmwxCk2:-499146009) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/3fb012c10cb809f2a34ffe30115888a91709f6ec/vaadin-date-picker-overlay.html#L32)):*
Done.

---

*[vaadin-date-picker-overlay.html, line 44 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KAsSed7R9Pt3XjrWSFE:-KB6Zd-R__K5AnZS-Z32:-499146009) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/3fb012c10cb809f2a34ffe30115888a91709f6ec/vaadin-date-picker-overlay.html#L44)):*
Done.

---

*[vaadin-date-picker-overlay.html, line 45 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KAp7f58IzppFXu6Ra3o-r1-45:-KB6ZqvhZ9-9ZqIobdxO:-499146009) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/3fb012c10cb809f2a34ffe30115888a91709f6ec/vaadin-date-picker-overlay.html#L45)):*
Done.

---

*[vaadin-date-picker-overlay.html, line 94 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KAsUDbccbECQtGR7VNu:-KB6_5v9olP-Ro7-0MMS:1855959810) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/3fb012c10cb809f2a34ffe30115888a91709f6ec/vaadin-date-picker-overlay.html#L94)):*
The animation should be a different patch

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44)*

**Comment by manolo** [comment]
Done. I did already but seems like I dint't push the correct amend

---

Review status: 2 of 3 files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-:-KB7V6F3RSFROYY6-13L:-2144781205)*

**Comment by tehapo** [comment]
Review status: 0 of 3 files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[vaadin-date-picker-overlay.html, line 43 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KBDS9Lpxojsq0ZUXy_k:-KBDS9Lpxojsq0ZUXy_l:-1931907849) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/19d41cd71d1f7215c66307b31fee08b93e81c2dc/vaadin-date-picker-overlay.html#L43)):*
You should also add `paper-button` as an explicit dependency to the `bower.json` file.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44)*

**Comment by tehapo** [comment]
LGTM, except for the missing `paper-button` dependency in `bower.json`.

---

Reviewed 1 of 1 files at r2, 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-:-KBDSbSbFnuuM3dKA_UN:-1834328594)*

**Comment by manolo** [comment]
Added

---

Review status: 1 of 4 files reviewed at latest revision, 5 unresolved discussions.

---

*[vaadin-date-picker-overlay.html, line 43 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KBDS9Lpxojsq0ZUXy_k:-KBH_brvKFa4IZLbRfB3:-499146009) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/19d41cd71d1f7215c66307b31fee08b93e81c2dc/vaadin-date-picker-overlay.html#L43)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-:-KBH_Z0ji1C_Mf5dUciI:-502253829)*

**Comment by tomivirkki** [comment]
*[bower.json, line 28 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KBH_wLS90tMWecp53ch:-KBH_wLTkgm0vjMX8zBO:-1959191985) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/1f1c985f7a11f935b52ab4a6994565b0b51f52ef/bower.json#L28)):*
should say paper-button I guess?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44)*

**Comment by manolo** [comment]
Review status: 1 of 4 files reviewed at latest revision, 6 unresolved discussions.

---

*[bower.json, line 28 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KBH_wLS90tMWecp53ch:-KBHaKQmVr0pN0186TT0:-989152395) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/1f1c985f7a11f935b52ab4a6994565b0b51f52ef/bower.json#L28)):*
Yep, Done

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44)*

**Comment by tomivirkki** [comment]
Review status: 1 of 4 files reviewed at latest revision, 8 unresolved discussions.

---

*[test/overlay.html, line 118 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KBHaS7C9VFADEkCV86e:-KBHaS7DDSU5Ava6lQT8:1155720436) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/1f1c985f7a11f935b52ab4a6994565b0b51f52ef/test/overlay.html#L118)):*
'tapping'

---

*[vaadin-date-picker-overlay.html, line 256 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KBHb5iNxXEcKLU_6LeN:-KBHb5iNxXEcKLU_6LeO:-1102843927) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/1f1c985f7a11f935b52ab4a6994565b0b51f52ef/vaadin-date-picker-overlay.html#L256)):*
Cancel button doesn't need to be a paper-button as it just closes the dropdown and doesn't have an icon. See the design at https://drive.google.com/a/vaadin.com/file/d/0BzkWbWvgmkIxWVItR3pOd0E2a28/view

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44)*

**Comment by tomivirkki** [comment]
Review status: 1 of 4 files reviewed at latest revision, 10 unresolved discussions.

---

*[vaadin-date-picker-overlay.html, line 36 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KBHbinDBRmh_vzrY2Py:-KBHbinEFk8rJV8915D8:-2137712073) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/1f1c985f7a11f935b52ab4a6994565b0b51f52ef/vaadin-date-picker-overlay.html#L36)):*
This change shouldn't affect the toolbar. Also, we're don't import iron-flex-layout

---

*[vaadin-date-picker-overlay.html, line 220 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KBHcPlczQsAV4iliLk1:-KBHcPlczQsAV4iliLk2:-772587904) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/1f1c985f7a11f935b52ab4a6994565b0b51f52ef/vaadin-date-picker-overlay.html#L220)):*
The ripple effect looks weird with padding:0 as it only partially covers the icon.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44)*

**Comment by platosha** [comment]
Review status: 1 of 4 files reviewed at latest revision, 10 unresolved discussions.

---

*[vaadin-date-picker-overlay.html, line 256 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KBHb5iNxXEcKLU_6LeN:-KBHcUPHX8XJgmYeZglv:1632985768) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/1f1c985f7a11f935b52ab4a6994565b0b51f52ef/vaadin-date-picker-overlay.html#L256)):*
The `paper-button` would still give us a nice ripple effect on mousedown/touchdown.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44)*

**Comment by tomivirkki** [comment]
*[vaadin-date-picker-overlay.html, line 256 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KBHb5iNxXEcKLU_6LeN:-KBHd60N_f0wBOdWtapw:305873316) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/1f1c985f7a11f935b52ab4a6994565b0b51f52ef/vaadin-date-picker-overlay.html#L256)):*
So it does. So just remove the icon then.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44)*

**Comment by platosha** [comment]
Review status: 1 of 4 files reviewed at latest revision, 8 unresolved discussions.

---

*[vaadin-date-picker-overlay.html, line 254 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KBHdMgelMU_wP6gQ_RD:-KBHdMgelMU_wP6gQ_RE:-486485802) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/1f1c985f7a11f935b52ab4a6994565b0b51f52ef/vaadin-date-picker-overlay.html#L254)):*
Despite that `today` and `cancel` button titles would be uppercased by `paper-button`'s styles, they are still titles. Would prefer them starting with a capital letter.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44)*

**Comment by tomivirkki** [comment]
*[vaadin-date-picker-overlay.html, line 254 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KBHdMgelMU_wP6gQ_RD:-KBHe6N20Z3rbYmyKMW3:1001137776) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/1f1c985f7a11f935b52ab4a6994565b0b51f52ef/vaadin-date-picker-overlay.html#L254)):*
And since paper-button does this, you should remove the existing "text-transform" rule from this component's #footer css

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44)*

**Comment by manolo** [comment]
Review status: 1 of 4 files reviewed at latest revision, 8 unresolved discussions.

---

*[test/overlay.html, line 118 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KBHaS7C9VFADEkCV86e:-KBHeXYjbYU0ptwEtSch:-499146009) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/1f1c985f7a11f935b52ab4a6994565b0b51f52ef/test/overlay.html#L118)):*
Done.

---

*[vaadin-date-picker-overlay.html, line 36 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KBHbinDBRmh_vzrY2Py:-KBHhsUieFW96rN7-biB:-499146009) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/1f1c985f7a11f935b52ab4a6994565b0b51f52ef/vaadin-date-picker-overlay.html#L36)):*
Done.

---

*[vaadin-date-picker-overlay.html, line 220 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KBHcPlczQsAV4iliLk1:-KBHef3WuBryEuhH_O6f:243171836) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/1f1c985f7a11f935b52ab4a6994565b0b51f52ef/vaadin-date-picker-overlay.html#L220)):*
Right, decreased padding to footer instead, so as the footer keeps the height.

---

*[vaadin-date-picker-overlay.html, line 254 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KBHdMgelMU_wP6gQ_RD:-KBHfqHRWXpKA28VWd7F:-499146009) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/1f1c985f7a11f935b52ab4a6994565b0b51f52ef/vaadin-date-picker-overlay.html#L254)):*
Done.

---

*[vaadin-date-picker-overlay.html, line 256 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KBHb5iNxXEcKLU_6LeN:-KBHhlWY9jNRHc6jYyAA:-499146009) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/1f1c985f7a11f935b52ab4a6994565b0b51f52ef/vaadin-date-picker-overlay.html#L256)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44)*

**Comment by tomivirkki** [comment]
*[test/overlay.html, line 118 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KBHaS7C9VFADEkCV86e:-KBHrFLQtyKtOKrYBXnZ:2129742234) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/1f1c985f7a11f935b52ab4a6994565b0b51f52ef/test/overlay.html#L118)):*
Still says taping

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44)*

**Comment by tomivirkki** [comment]
Reviewed 1 of 1 files at r5.
Review status: 2 of 4 files reviewed at latest revision, 8 unresolved discussions.

---

*[vaadin-date-picker-overlay.html, line 220 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KBHcPlczQsAV4iliLk1:-KBHs7ei0TzwZlV0cfcm:-1344356843) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/1f1c985f7a11f935b52ab4a6994565b0b51f52ef/vaadin-date-picker-overlay.html#L220)):*
Looks better now but the buttons are too low now. A bit more padding-bottom should fix it

---

*[vaadin-date-picker-overlay.html, line 256 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KBHb5iNxXEcKLU_6LeN:-KBHsY2pueRJD48WEbH9:1847459555) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/1f1c985f7a11f935b52ab4a6994565b0b51f52ef/vaadin-date-picker-overlay.html#L256)):*
Please take a look at the design document I linked. There is an icon on the other button.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44)*

**Comment by manolo** [comment]
Review status: 2 of 4 files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---

*[test/overlay.html, line 118 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KBHaS7C9VFADEkCV86e:-KBI75Eq9X3B7Qboz8fX:-499146009) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/1f1c985f7a11f935b52ab4a6994565b0b51f52ef/test/overlay.html#L118)):*
Done.

---

*[vaadin-date-picker-overlay.html, line 220 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KBHcPlczQsAV4iliLk1:-KBI70gC8I0jv8ujJ7S6:-499146009) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/1f1c985f7a11f935b52ab4a6994565b0b51f52ef/vaadin-date-picker-overlay.html#L220)):*
Done.

---

*[vaadin-date-picker-overlay.html, line 256 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44#-KBHb5iNxXEcKLU_6LeN:-KBI73jvChewy95mkuQm:-499146009) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/1f1c985f7a11f935b52ab4a6994565b0b51f52ef/vaadin-date-picker-overlay.html#L256)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44)*

**Comment by tomivirkki** [comment]
Reviewed 2 of 2 files at r7.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44)*

**Comment by tomivirkki** [comment]
Reviewed 1 of 1 files at r2, 2 of 2 files at r8.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/44)*

**Pull request merged**